File: 프로그래머스/1/155652. 둘만의 암호/둘만의 암호.py
def solution(s, skip, index):
    answer = []
    alphabet = [chr(i) for i in range(97, 123)]
    for c in skip:
        alphabet.remove(c)

    for c in s:
        idx = (alphabet.index(c) + index) % len(alphabet)
        answer.append(alphabet[idx])
    
    answer = ''.join(answer)
    
    return answer

File: 프로그래머스/1/155652. 둘만의 암호/둘만의 암호.swift
import Foundation

func solution(_ s:String, _ skip:String, _ index:Int) -> String {
    let skipped = Array("abcdefghijklmnopqrstuvwxyz").filter { !skip.contains($0) }
    var result = ""
    
    for c in s {
        let idx = skipped.firstIndex(of: c)!
        let newIdx = (idx + index ) % skipped.count
        result += String(skipped[newIdx])
    }
    
    return result
}